client/app/visualizations/counter/Editor.jsx

```diff
@@ -0,0 +1,236 @@
+          <label htmlFor="counter-formatting-thousands-separator">Formatting Thousands Separator</label>
+        </Grid.Col>
+        <Grid.Col span={12}>
+          <Input
+            id="counter-formatting-thousands-separator"
+            className="w-100"
+            data-test="Counter.Formatting.ThousandsSeparator"
+            defaultValue={options.stringThouSep}
+            disabled={!inputsEnabled}
+            onChange={e => onOptionsChange({ stringThouSep: e.target.value })}
+          />
+        </Grid.Col>
+      </Grid.Row>
+
+      <Grid.Row type="flex" align="middle" className="m-b-10">
+        <Grid.Col span={12}>
+          <label htmlFor="counter-formatting-string-prefix">Formatting String Prefix</label>
+        </Grid.Col>
+        <Grid.Col span={12}>
+          <Input
+            id="counter-formatting-string-prefix"
+            className="w-100"
+            data-test="Counter.Formatting.StringPrefix"
+            defaultValue={options.stringPrefix}
+            disabled={!inputsEnabled}
+            onChange={e => onOptionsChange({ stringPrefix: e.target.value })}
+          />
+        </Grid.Col>
+      </Grid.Row>
+
+      <Grid.Row type="flex" align="middle" className="m-b-10">
+        <Grid.Col span={12}>
+          <label htmlFor="counter-formatting-string-suffix">Formatting String Suffix</label>
+        </Grid.Col>
+        <Grid.Col span={12}>
+          <Input
+            id="counter-formatting-string-suffix"
+            className="w-100"
+            data-test="Counter.Formatting.StringSuffix"
+            defaultValue={options.stringSuffix}
+            disabled={!inputsEnabled}
+            onChange={e => onOptionsChange({ stringSuffix: e.target.value })}
+          />
+        </Grid.Col>
+      </Grid.Row>
+
+      <label className="d-flex align-items-center" htmlFor="counter-format-target-value">
+        <Switch
+          id="counter-format-target-value"
+          data-test="Counter.Formatting.FormatTargetValue"
+          defaultChecked={options.formatTargetValue}
+          onChange={formatTargetValue => onOptionsChange({ formatTargetValue })}
+        />
+        <span className="m-l-10">Format Target Value</span>
+      </label>
+    </React.Fragment>
+  );
+}
+
+FormatSettings.propTypes = EditorPropTypes;
+
+export default function Editor(props) {
+  const { options, onOptionsChange } = props;
+
+  const optionsChanged = (newOptions) => {
+    onOptionsChange(merge({}, options, newOptions));
+  };
+
+  return (
+    <Tabs animated={false}>
+      <Tabs.TabPane key="general" tab={<span data-test="Counter.EditorTabs.General">General</span>}>
+        <GeneralSettings {...props} onOptionsChange={optionsChanged} />
+      </Tabs.TabPane>
+      <Tabs.TabPane key="format" tab={<span data-test="Counter.EditorTabs.Formatting">Format</span>}>
+        <FormatSettings {...props} onOptionsChange={optionsChanged} />
+      </Tabs.TabPane>
+    </Tabs>
+  );
+}
+
+Editor.propTypes = EditorPropTypes;
```

Comment:
@arikfr wdyt of having a link to the counter video from here?
(Actually, linking to a help doc or blog post with the video in it would be better)

Reply:
I agree that having a help trigger here makes sense. :+1: But I guess it should be a global thing for all visualizations, which we enable when we have a relevant page?

Reply:
We can embed the video here then https://redash.io/help/user-guide/visualizations/visualization-types#Counter